appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java

```diff
@@ -1701,15 +1710,13 @@ public String SaveAs(String fileName) {
   private String saveFile(File file, Bitmap.CompressFormat format, String method) {
     try {
       boolean success = false;
-      FileOutputStream fos = new FileOutputStream(file);
-      // Don't cache, in order to save memory.  It seems unlikely to be used again soon.
-      Bitmap bitmap = (view.completeCache == null ? view.buildCache() : view.completeCache);
-      try {
+      // Don't cache, in order to save memory.
+      // It seems unlikely to be used again soon.
+      try (FileOutputStream fos = new FileOutputStream(file)) {
```

Comment:
Note that try-with-resources is only supported on Android 7 and later. Making this change will break saving the canvas on versions of Android prior to Android 7.

Reply:
I have reverted the unnecessary changes.